tools/metatoenv.py

```diff
@@ -127,15 +127,22 @@ def _parse_yaml(text):
     return out
 
 
-def _parse_py_version(ver):
+def _parse_py_version(ver, drop_minor, drop_micro):
     """
-    If the input version string has no '.' character, e.g. `38`, then change it to
-    `3.8`.
     Return a version parsed through the `packaging.version` tool.
+    If `drop_minor` is `True`, we drop the minor version (set it to 0).
+    If `drop_micro` is `True`, we drop the micro version (set it to 0).
     """
-    if (len(ver) > 1) and '.' not in ver:
+    # If the input version string has has length 2 and no '.' character, e.g. `38`,
+    # then change it to `3.8`.
+    if (len(ver) > 1) and ('.' not in ver):
         ver = f'{ver[0]}.{ver[1:]}'
-    return version.parse(ver)
+    out = version.parse(ver)
+    if drop_micro:
+        out = version.parse(f'{out.major}.{out.minor}')
+    if drop_minor:
+        out = version.parse(f'{out.major}')
```

Comment:
Do we need to ever drop the minor version? Seems like this would make the check pretty much useless. Unless you expect to check for python 4 at some point.

Reply:
yeah, i think I will remove..